Content.Shared/Projectiles/SharedProjectileSystem.cs

```diff
@@ -117,12 +122,39 @@ private void Embed(EntityUid uid, EntityUid target, EntityUid? user, EmbeddableP
         Dirty(uid, component);
     }
 
-    public void UnEmbed(EntityUid uid, EmbeddableProjectileComponent? component, EntityUid? user = null)
+    /// <summary>
+    ///  Makes the specified entity not be embedded in whatever it's embedded in. In the case that the specified entity
+    ///  is not embedded in anything, this function does nothing.
+    /// </summary>
+    /// <param name="uid">The entity to make no longer embedded</param>
+    /// <param name="component"><paramref name="uid"/>'s projectile component</param>
+    /// <param name="removingUser">The entity which is removing the embedded entity. If not null, we'll try to put the
+    /// embedded object in its hands. If null, there's no specific remover, eg. if the embeddee object is destroyed.</param>
+    public void UnEmbed(EntityUid uid, EmbeddableProjectileComponent? component, EntityUid? removingUser = null)
     {
         if (!Resolve(uid, ref component))
             return;
 
         var xform = Transform(uid);
+
+        // Check that the projectile's parent has any embedded projectiles and that this projectile is one of them.
+        if (!(TryComp<HasProjectilesEmbeddedComponent>(xform.ParentUid, out var entitiesEmbeddedInParent) &&
+              entitiesEmbeddedInParent.EmbeddedProjectiles.Contains(uid)))
+            return;
+
+        // Remove `entity` from the parent's embedded projectiles, and clean up the parent's embedding component if it's empty.
+        entitiesEmbeddedInParent.EmbeddedProjectiles.Remove(uid);
+        if (entitiesEmbeddedInParent.EmbeddedProjectiles.Count == 0)
+        {
+            EntityManager.RemoveComponent<HasProjectilesEmbeddedComponent>(xform.ParentUid);
+        }
+
+        if (component.DeleteOnRemove)
+        {
+            QueueDel(uid);
+            return;
+        }
```

Comment:
@ScarKy0 , can you please verify that this is alright to put here? Basically if Recall is used on something with `DeleteOnRemove`, is it reasonable for the item to get deleted rather than recalled? Or perhaps would something recalled never be `DeleteOnRemove`?

I'd just so hate to have to add a flag arg for this that turns on or off functionality of the component...